src/lib/baseCommand.ts

```diff
@@ -11,11 +10,16 @@ import isCI from './isCI';
 import { debug, info, warn } from './logger';
 import loginFlow from './loginFlow';
 
-export type CommandOptions<T> = T & {
-  github?: boolean;
-  key?: string;
+export type CommandOptions<T> = ZeroAuthCommandOptions<T> | AuthenticatedCommandOptions<T>;
+
+export type AuthenticatedCommandOptions<T> = Omit<ZeroAuthCommandOptions<T>, 'key'> & {
+  key: string;
   version?: string;
-} & CommandLineOptions;
+};
+
+export type ZeroAuthCommandOptions<T> = T & {
+  github?: boolean;
+} & { key?: never };
```

Comment:
split out our command options for commands that do and do not require auth so we could assert that commands with zero auth do **not** accept a key parameter. TS is cool sometimes